rancher/rancher
# Bump rancher/machine to v0.15.0-rancher103

## Issue: 

 
https://github.com/rancher/rancher/issues/42308
SURE-3285
Internal design docs


## Problem

 
Add the support for setting graceful shutdown timeout for vSphere node-driver RKE1/RKE2/K3s clusters 

## Solution

 
The implementation has been done in the rancher/machine repo. Details can be found in the internal design doc. 

This PR is to bring the new version of rancher-machine into Rancher. 

## Testing


To test the new feature, we need to create vSphere node-driver RKE1/RKE2/K3s clusters and utilize the new `gracefulShutdownTimeout` option on either `nodeTemplate` for RKE1 or `vmwarevsphereconfigs` for RKE2/K3s. 

## Engineering Testing
### Manual Testing


With the new version of rancher-machine, the new field `gracefulShutdownTimeout` should be available for vSphere Node template (RKE1) and vmwarevsphereconfigs (RKE2/K3s)

To prove that, I run rancher in my local dev env by pointing rancher-machine to another local dev build, and seeing the following in the local cluster:

```
// For RKE2/K3s
> k get crd vmwarevsphereconfigs.rke-machine-config.cattle.io -o yaml  | grep grace
          gracefulShutdownTimeout:
              a graceful shutdown for a vSphere virtual machine. A force destroy will
```

```
// For RKE1, create a vShpere node Template via UI first  
> k -n cattle-global-nt get nodetemplate  nt-mps4h -o yaml | grep grace
  gracefulShutdownTimeout: "0"
```

### Automated Testing

This should be covered by existing automated tests on the QA side for provisioning clusters if there are any for vSphere node-driver clusters. 

## QA Testing Considerations

 
- Create node-drive vSphere cluster and set the `gracefulShutdownTimeout` option. 
- Edit and update existing node-drive vSphere cluster to use the new option 

### Regressions Considerations

Provisioning/updating node-drive vSphere cluster fails.
